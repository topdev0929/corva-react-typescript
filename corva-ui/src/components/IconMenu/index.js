import { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { IconButton, Menu, withStyles, Tooltip } from '@material-ui/core';
import { MoreVert } from '@material-ui/icons';

const muiStyles = theme => ({
  iconButton: {
    padding: '10px',
    color: theme.palette.primary.text6,
    '&:hover': {
      color: theme.palette.primary.text1,
    },
    paper: {},
  },
});

function IconMenu({
  classes,
  // IconButtonComponentProps
  'data-testid': testId,
  iconButtonStyle,
  color,
  disableRipple,
  className,
  IconElement,
  IconButtonComponent,
  disabled,
  iconButtonProps,
  // MenuProps
  children,
  anchorOrigin,
  transformOrigin,
  marginThreshold,
  menuProps,
  // Tooltip
  tooltipProps,
  paperClassName,
}) {
  const [anchorEl, setAnchorEl] = useState(null);
  const isOpen = !!anchorEl;

  /*
    A fix of the issue: https://corvaqa.atlassian.net/browse/DC-3713

    When Material Menu is rendered, at first it's displayed at the top of the container
    and only then moves to the anchor element. When the focus is enabled, and the element is
    inside of an iframe but outside the view - browsers scroll the page to it. Because of this, we set focus in async
    manner, to wait for correct position - and then set focus.

    https://github.dev/mui/material-ui/blob/v4.x/packages/material-ui/src/Popover/Popover.js#L323-L331
  */
  const [isMenuAutofocus, setIsMenuAutofocus] = useState(false);
  useEffect(() => {
    let timerId;

    if (isOpen) {
      setTimeout(() => {
        timerId = setIsMenuAutofocus(isOpen);
        // Why 500? The lower delay doesn't actually work, so this is the lowest value I could get to make it work
      }, 500);
    }

    return () => {
      if (timerId) {
        clearTimeout(timerId);
      }
    };
  }, [isOpen]);

  const handleClose = event => {
    event.stopPropagation();
    setAnchorEl(null);
    setIsMenuAutofocus(false);
  };
  const handleOpen = event => {
    event.stopPropagation();
    event.preventDefault();
    setAnchorEl(event.currentTarget);
  };

  const iconButtonElement = (
    <IconButtonComponent
      aria-label="IconMenu"
      aria-owns={isOpen ? 'icon-menu' : undefined}
      aria-haspopup="true"
      data-testid={testId}
      onClick={handleOpen}
      classes={{ root: classes.iconButton }}
      className={className}
      style={iconButtonStyle}
      color={color}
      disableRipple={disableRipple}
      disabled={disabled}
      {...iconButtonProps}
    >
      {IconElement}
    </IconButtonComponent>
  );

  return (
    <>
      {tooltipProps ? <Tooltip {...tooltipProps}>{iconButtonElement}</Tooltip> : iconButtonElement}
      <Menu
        id="icon-menu"
        anchorEl={anchorEl}
        open={isOpen}
        disableAutoFocus={!isMenuAutofocus}
        autoFocus={isMenuAutofocus}
        onClose={handleClose}
        onClick={handleClose}
        anchorOrigin={anchorOrigin}
        transformOrigin={transformOrigin}
        classes={{ paper: classes.paper }}
        elevation={0}
        marginThreshold={marginThreshold}
        PaperProps={{
          className: classes[paperClassName],
        }}
        {...menuProps}
      >
        {children}
      </Menu>
    </>
  );
}

IconMenu.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.array]),
  classes: PropTypes.shape({
    iconButton: PropTypes.string,
    paper: PropTypes.string,
  }),
  className: PropTypes.string,
  color: PropTypes.string,
  IconElement: PropTypes.element,
  // eslint-disable-next-line react/forbid-prop-types
  IconButtonComponent: PropTypes.any,
  iconButtonStyle: PropTypes.shape({}),
  anchorOrigin: PropTypes.shape({
    horizontal: PropTypes.oneOf(['left', 'center', 'right']),
    vertical: PropTypes.oneOf(['top', 'center', 'bottom']),
  }),
  transformOrigin: PropTypes.shape({
    horizontal: PropTypes.oneOf(['left', 'center', 'right']),
    vertical: PropTypes.oneOf(['top', 'center', 'bottom']),
  }),
  disableRipple: PropTypes.bool,
  'data-testid': PropTypes.string,
  marginThreshold: PropTypes.number,
  disabled: PropTypes.bool,
  tooltipProps: PropTypes.shape({}),
  menuProps: PropTypes.shape({}),
  iconButtonProps: PropTypes.shape({}),
  paperClassName: PropTypes.string,
};

IconMenu.defaultProps = {
  children: undefined,
  classes: {},
  className: undefined,
  color: undefined,
  'data-testid': undefined,
  IconElement: <MoreVert />,
  iconButtonStyle: undefined,
  IconButtonComponent: IconButton,
  disableRipple: false,
  anchorOrigin: { horizontal: 'right', vertical: 'bottom' },
  transformOrigin: { horizontal: 'right', vertical: 'bottom' },
  marginThreshold: undefined,
  disabled: false,
  tooltipProps: undefined,
  menuProps: {},
  iconButtonProps: {},
  paperClassName: '',
};

export default withStyles(muiStyles)(IconMenu);
