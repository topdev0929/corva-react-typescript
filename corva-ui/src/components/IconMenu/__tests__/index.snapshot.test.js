import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import IconMenu from '../index';

describe('Snapshot', () => {
  describe('IconMenu', () => {
    it('default props', () => {
      const tree = render(<IconMenu />).baseElement;
      expect(tree).toMatchSnapshot();
    });

    it('classes', () => {
      const tree = render(<IconMenu classes={{ iconButton: 'iconButton', paper: 'paper' }} />).baseElement;
      expect(tree).toMatchSnapshot();
    });

    it('className', () => {
      const tree = render(<IconMenu className='className' />).baseElement;
      expect(tree).toMatchSnapshot();
    });

    it('color', () => {
      const tree = render(<IconMenu color='inherit' />).baseElement;
      expect(tree).toMatchSnapshot();
    });

    it('IconElement', () => {
      const tree = render(<IconMenu IconElement={<div>IconElement</div>} />).baseElement;
      expect(tree).toMatchSnapshot();
    });

    it('IconButtonComponent', () => {
      const tree = render(<IconMenu IconButtonComponent={() => (<div>IconButtonComponent</div>)} />).baseElement;
      expect(tree).toMatchSnapshot();
    });

    it('iconButtonStyle', () => {
      const tree = render(<IconMenu iconButtonStyle={{ float: 'right', marginBottom: 2 }} />).baseElement;
      expect(tree).toMatchSnapshot();
    });

    it('data-testid', () => {
      const tree = render(<IconMenu data-testid='data-testid' />).baseElement;
      expect(tree).toMatchSnapshot();
    });

    it('marginThreshold', () => {
      const tree = render(<IconMenu marginThreshold={42} />).baseElement;
      expect(tree).toMatchSnapshot();
    });

    describe('children', () => {
      it('string', () => {
        const tree = render(<IconMenu>{'string'}</IconMenu>).baseElement;
        expect(tree).toMatchSnapshot();
      });

      it('node', () => {
        const tree = render(<IconMenu>
          <div>node</div>
        </IconMenu>).baseElement;
        expect(tree).toMatchSnapshot();
      });
    });

    describe('disableRipple', () => {
      it('true', () => {
        const tree = render(<IconMenu disableRipple={true} />).baseElement;
        expect(tree).toMatchSnapshot();
      });

      it('false', () => {
        const tree = render(<IconMenu disableRipple={false} />).baseElement;
        expect(tree).toMatchSnapshot();
      });
    });

    describe('handle actions', () => {
      it('handleOpen', () => {
        const { getByTestId, baseElement } = render(<IconMenu data-testid='test1' />);
        userEvent.click(getByTestId('test1'));
        expect(baseElement).toMatchSnapshot();
      });

      it('handleClose by click', () => {
        const { getByRole, baseElement, getByTestId } = render(<IconMenu data-testid='test1' />);
        userEvent.click(getByTestId('test1'));
        userEvent.click(getByRole('menu'));
        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});

