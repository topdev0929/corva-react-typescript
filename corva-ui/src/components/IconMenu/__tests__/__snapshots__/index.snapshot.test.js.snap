// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot IconMenu IconButtonComponent 1`] = `
<body>
  <div>
    <div>
      IconButtonComponent
    </div>
  </div>
</body>
`;

exports[`Snapshot IconMenu IconElement 1`] = `
<body>
  <div>
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-5"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <div>
          IconElement
        </div>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Snapshot IconMenu children node 1`] = `
<body>
  <div>
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-11"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Snapshot IconMenu children string 1`] = `
<body>
  <div>
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-10"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Snapshot IconMenu className 1`] = `
<body>
  <div>
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-3 className"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Snapshot IconMenu classes 1`] = `
<body>
  <div>
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-2 iconButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Snapshot IconMenu color 1`] = `
<body>
  <div>
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-4 MuiIconButton-colorInherit"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Snapshot IconMenu data-testid 1`] = `
<body>
  <div>
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-8"
      data-testid="data-testid"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Snapshot IconMenu default props 1`] = `
<body>
  <div>
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Snapshot IconMenu disableRipple false 1`] = `
<body>
  <div>
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-13"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Snapshot IconMenu disableRipple true 1`] = `
<body>
  <div>
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-12"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
    </button>
  </div>
</body>
`;

exports[`Snapshot IconMenu handle actions handleClose by click 1`] = `
<body
  style=""
>
  <div>
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-15"
      data-testid="test1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>
  <div
    aria-hidden="true"
    class="MuiPopover-root"
    id="icon-menu"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiMenu-paper MuiPopover-paper MuiPaper-elevation0 MuiPaper-rounded"
      style="opacity: 0; transform: scale(0.75, 0.5625); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;"
      tabindex="-1"
    >
      <ul
        class="MuiList-root MuiMenu-list MuiList-padding"
        role="menu"
        tabindex="-1"
      />
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Snapshot IconMenu handle actions handleOpen 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      aria-owns="icon-menu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-14"
      data-testid="test1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>
  <div
    class="MuiPopover-root"
    id="icon-menu"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: transparent;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiMenu-paper MuiPopover-paper MuiPaper-elevation0 MuiPaper-rounded"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;"
      tabindex="-1"
    >
      <ul
        class="MuiList-root MuiMenu-list MuiList-padding"
        role="menu"
        tabindex="-1"
      />
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Snapshot IconMenu iconButtonStyle 1`] = `
<body>
  <div>
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-7"
      style="float: right; margin-bottom: 2px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`Snapshot IconMenu marginThreshold 1`] = `
<body>
  <div>
    <button
      aria-haspopup="true"
      aria-label="IconMenu"
      class="MuiButtonBase-root MuiIconButton-root IconMenu-iconButton-9"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
